crates/matrix-sdk-ui/src/timeline/event_item/local.rs

```diff
@@ -65,7 +65,10 @@ impl LocalEventTimelineItem {
 #[derive(Clone, Debug)]
 pub enum EventSendState {
     /// The local event has not been sent yet.
-    NotSentYet,
+    NotSentYet {
+        /// The progress of the sending operation, if any is available.
+        progress: Option<EventSendProgress>,
+    },
```

Comment:
Based on other comments: we might prefer to have a new variant `NotUploadedYet`, for medias. The `progress` field isn't used, for non-media uploads.